chapters/feature-selection.qmd

```diff
@@ -0,0 +1,525 @@
+- Boosted trees (via lightGBM)
+- Cubist
+- Generalized additive models (GAMs)
+- K-nearest neighbors (KNN)
+- Multivariate adaptive regression splines (MARS)
+- Penalized linear regression
+- Random forest
+- RuleFit
+- Single-layer neural networks
+- Support vector machines (radial basis function kernel)
+
+These models are described in detail in subsequent chapters. The colors of the points/lines signifies whether the model automatically removes unused predictors from the model equation (see @sec-automatic-selection below). 
+
+```{r}
+#| label: fig-irrelevant-predictors
+#| echo: false
+#| out-width: 70%
+#| fig-width: 6
+#| fig-height: 5
+#| fig-cap: For different regression models, the percent increase in RMSE is shown as a function of the number of additional non-informative predictors.
+
+noise_simulation_results_pct %>%
+  mutate(
+    `Automatic Selection` = ifelse(fs, "yes", "no")
+  ) %>%
+  ggplot(aes(num_extra, increase)) +
+  geom_hline(yintercept = 0, col = "green", alpha = 1 / 2) +
+  geom_point(aes(col = `Automatic Selection`)) +
+  geom_line(aes(col = `Automatic Selection`)) +
+  facet_wrap(~label) +
+  labs(x = "# Noise Features", y = "RMSE Increase") +
+  scale_y_continuous(labels = scales::percent)
+```
+
+The results show that a few models, K-nearest neighbors, neural networks, and support vector machines, have severely degraded performance as the number of predictors increases. This is most likely due to the use of cross- and/or dot-products of the predictor columns in their calculations. The extra predictors add significant noise to these calculations, and that noise propagates in a way that inhibits the models from accurately determining the underlying relationship with the outcome. 
+
+However, several other models use these same calculations without being drastically affected. This is due to the models **automatically performing feature selection** during model training. Let’s briefly describe two clusters of models that were largely unaffected by the extra columns. 
+
+ 1. Penalized linear regression, generalized additive models, and RuleFit add a penalty to their calculations, restricting the model parameters from becoming abnormally large unless the underlying data warrant a large coefficient. This effectively reduces the impact of extra predictors by keeping their corresponding model coefficient at or near zero. 
+
+ 2. MARS and Cubist selectively include predictors in their regression equations, only including those specifically selected as having value. 
+
+The other unaffected models are tree-based. As seen in @fig-reg-tree, these types of models make different rules by selecting predictors to _split_ the data. If a predictor was not used in any split, it is functionally independent of the outcome in the prediction equation. This provides some insensitivity to the presence of non-informative predictors, although there is slight degradation in RMSE as the number of columns increases. 
```

Comment:
I think this section could help for a slight bit of reemphasis. It says "describe two clusters of models" then followed by 1., 2. and "The other". Maybe 3 clusters of models?